lib/internal/util.js

```diff
@@ -262,6 +262,22 @@ function promisify(original) {
 
 promisify.custom = kCustomPromisifiedSymbol;
 
+function promisifyAll(original) {
+  let type = typeof original
+  let obj = {}
+  if (type == "object") {
+      var properties = !original.length ? Object.keys(original) : original
```

Comment:
Using `Object.keys` here means that only an object's own properties are promisified - functions in the prototype chain will not be included.